admin/server/auth.py

```
@@ -154,18 +292,48 @@ def login_admin(email: str, password: str):
+    # one client cannot exceed ADMIN_LOGIN_MAX_FAILURES verifications.
+    remaining, marker = _reserve_login_attempt(key)
+    if remaining > 0:
+        raise AdminException(f"Too many failed login attempts. Retry in {remaining}s.", 429)
+
+    try:
+        users = UserService.query(email=email)
+        if not users:
+            raise UserNotFoundError(email)
+        try:
+            decrypted = decrypt(password)
+        except Exception:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)